### ISSUE TITLE
Error sending email: Expected response code 220 but got an empty response

### ISSUE THREAD

**Issue by piotrpog** [craftcms/contact-form#170]
When i try to send mail (using ajax), plugin returns message indicating that message was sent - `success: true`.

However no email is sent and in error log i get this message:

```
2019-06-26 02:35:28 [-][1][-][warning][application] Error sending email: Expected response code 220 but got an empty respons
```
Mail is supposed to be sent using server sendmail functionality - nothing was overwritten in `config/app.php` file.

I also tested if sendmail works using simple phpmailer script - and it works correctly. So issue seems to be with contact form plugin.

If you want, i can send you a log file.

**Comment by brandonkelly** [comment]
Is Craft able to send other emails, such as password reset emails?

**Comment by watarutmnh** [comment]
I have a similar problem.
I got this error when I send form using jQuery post method.
`Failed to load resource: the server responded with a status of 500 ()`
I noticed that It will be triggered when I set environment variable to fromEmail option in Setting > Email.
It be able to send a mail but fail to get response data.

**Comment by amityweb** [comment]
I get this when using Sendmail. I am using SMTP,  but I may have other issues with SMTP, it seems to send the email even though its spam and not show an error. I want to use Sendmail to test, but I get this "Expected response code 220 but got an empty response"

**Comment by timack** [comment]
I had this proplem to running Craft CMS on Docksal with [MailHog](https://docs.docksal.io/tools/mailhog/), we were able to resolve it by switching transport type from Sendmail to SMTP configured as below: 

<img width="610" alt="image" src="https://user-images.githubusercontent.com/1098068/73034990-803c8100-3eab-11ea-9a95-9fff2b6ef3ef.png">

**Comment by janhenckens** [comment]
Running into the same issue on a site that uses Sendmail for e-mail delivery.

**Comment by AugustMiller** [comment]
Cleaning up some old issues—this seems like a number of different, environment-specific problems that are out of the scope of Contact Form.

Our Knowledge Base has an article on [Troubleshooting Email Errors](https://craftcms.com/knowledge-base/troubleshooting-email-errors#dealing-with-email-delivery-issues).

**Comment by amityweb** [comment]
> Cleaning up some old issues—this seems like a number of different, environment-specific problems that are out of the scope of Contact Form.
> 
> Our Knowledge Base has an article on [Troubleshooting Email Errors](https://craftcms.com/knowledge-base/troubleshooting-email-errors#dealing-with-email-delivery-issues).

Why dont you just say this was 6 years ago and its all changed since then?? The above is definitely an issue with the code because every single other PHP based system that sends email works and this one didn't. So not really out of scope. Seems a lazy way to close it off. Anyway, best no one even uses this now its so old.

**Comment by AugustMiller** [comment]
Unfortunately, yours is the first [confirmation of Brandon’s question](#issuecomment-509005997) in all this time that a Craft application was able to send _other_ system messages (if I understand this correctly):

> …every single other PHP based system that sends email works and this one didn't…

The initial report only mentioned a “phpmailer” script… without knowing how that differs from the sendmail command configured in Craft, or having any information about the sendmail/SMTP capabilities of the environments, we don't really have anything to go on. 

To make sure everyone here is addressed properly:

- “I may have other issues with SMTP” ([Comment](https://github.com/craftcms/contact-form/issues/170#issuecomment-534079313)) — I'm considering this a potential “environmental difference” that we'd need to iron out
- “it seems to send the email even though its spam and not show an error” ([Comment](https://github.com/craftcms/contact-form/issues/170#issuecomment-534079313)) — This sounds like a different issue (#95 or #175)? Nothing to do with sendmail/SMTP, and just a (controversial!) behavioral/design choice that was made at the very beginning.
- “I got this error when I send form using jQuery post method.” ([Comment](https://github.com/craftcms/contact-form/issues/170#issuecomment-509078916))— Unrelated/off-topic.
- "…we were able to resolve it by switching transport type from Sendmail to SMTP…" ([Comment](https://github.com/craftcms/contact-form/issues/170#issuecomment-577938666)) — Evidence of additional “environmental differences,” like processes not being allowed to communicate on some ports (common SMTP ports are [often blocked by hosts](https://craftcms.com/knowledge-base/why-doesnt-craft-send-emails))
- "…best no one even uses this now its so old…" (Comment) — Agreed! Based support volume per mailer/adapter, we have recommended against sendmail for many years. I acted alone in closing this issue, but I believe the rest of the team would agree that propping up sendmail usage benefits almost no one.

> Why dont you just say this was 6 years ago and its all changed since then?

I think the more accurate conclusion here is actually that _nothing_ has changed in that time! That doesn't make your criticism invalid, though—I would love to get to the bottom of every issue, but at the end of the day, we have to apply ourselves in the way that makes the greatest impact. To that end, my issues “cleanup” follows a few weeks of work on a [4.x version](https://github.com/craftcms/contact-form/tree/4.x), which addresses almost all of the other outstanding bugs and feature requests.

**Comment by amityweb** [comment]
Just saying its 6 years do you need to? Does the issue exist still? Whoever had issues probably found alternative solutions by now. I only commented though just because the closure message was quite dismissive somehow suggesting it was our server causing the issue which I disagree with! Either way I personally dont care about this issue anymore as we got around it, 6 years ago! :laughing: Sorry to comment on it, I often get these notices years later after reporting issues and it bugs me :laughing:
